File: src/fsharp/Luke01/Program.fs
﻿open System
open System.IO

[<EntryPoint>]
let main argv =
    let nums =
        File.ReadAllText("numbers.txt").Split(',')
        |> Array.toList
        |> List.map (fun s -> s.Trim())
        |> List.map int
        
    [1..100000]
    |> Seq.except nums
    |> Seq.head
    |>  printfn "%i"
    |> ignore
    
    0

File: src/CSharp/Luke01/Program.cs
using System;
using System.IO;
using System.Linq;

namespace Luke01
{
	public static class Program
	{
		public static void Run()
		{
			Console.WriteLine(
				File.ReadAllText("numbers.txt")
					.Split(',')
					.Select(s => s.Trim())
					.Select(s => int.Parse(s))
					.OrderBy(i => i)
					.Where(((i, idx) => idx + 1 != i))
					.Select(i => i - 1)
					.FirstOrDefault());

			Console.WriteLine(
				Enumerable.Range(1, 100_000)
					.Except(
						File.ReadAllText("numbers.txt")
							.Split(',')
							.Select(s => s.Trim())
							.Select(s => int.Parse(s)))
					.FirstOrDefault());
		}
	}
}
